src/components/modal/Modal.tsx

```diff
@@ -0,0 +1,80 @@
+import {SubtleButton} from '@utrecht/component-library-react';
+import {useEffect, useId, useRef} from 'react';
+import {useIntl} from 'react-intl';
+
+import Icon from '@/components/icons';
+
+import './Modal.scss';
+
+// @TODO is this needed? Or does the HTMl dialog handle this natively?
+// Copied logic from the current SDK implementation.
+// Used to prevent scrolling in the page, when the modal is open.
+const usePreventScroll = (open: boolean): void => {
+  useEffect(() => {
+    if (open) {
+      document.body.style.overflow = 'hidden';
+    } else {
+      document.body.style.overflow = 'unset';
+    }
+    return () => {
+      document.body.style.overflow = 'unset';
+    };
+  }, [open]);
+};
+
+export interface ModalProps {
+  isOpen?: boolean;
+  title?: React.ReactNode;
+  closeModal: () => void;
+  children: React.ReactNode;
+}
+
+const Modal: React.FC<ModalProps> = ({isOpen = false, title, closeModal, children}) => {
+  usePreventScroll(isOpen);
+  const id = useId();
+  const intl = useIntl();
+  const modalRef = useRef<HTMLDialogElement>(null);
+
+  const closeButtonLabel = intl.formatMessage({
+    description: 'Modal close button title',
+    defaultMessage: 'Close modal',
+  });
+
+  // To open the dialog as a modal, we need to trigger its javascript functions.
+  useEffect(() => {
+    const modal = modalRef.current;
+    if (!modal) {
+      return;
+    }
+
+    isOpen ? modal.showModal() : modal.close();
+  }, [isOpen]);
+
+  return (
+    <dialog
+      className="modal"
```

Comment:
```suggestion
      className="openforms-modal"
```
let's stick to our existing class name pattern and properly namespace things